app/jobs/base_job.rb

```diff
@@ -5,67 +5,48 @@
 ###
 
 class BaseJob < ApplicationJob
-  STARTING_PATH = File.realpath(FileUtils.pwd)
   include NotifierConfig
 
-  if ENV['ECS'] != 'true'
-    # When called through Active::Job, uses this hook
-    before_perform do |job|
-      if STARTING_PATH != expected_path || ! File.exist?('config/exception_notifier.yml')
-        msg = "Started dir is `#{STARTING_PATH}`\nCurrent dir is `#{expected_path}`\nExiting in order to let systemd restart me in the correct directory."
-        notify_on_restart(msg)
+  attr_accessor :start_time
 
-        unlock_job!(job.job_id) if job.respond_to? :job_id
+  if ENV['EKS'] == 'true'
+    # I can't get this to work correctly from the delayed_job initializer
+    Rails.logger.info 'Registering prometheus metrics for delayed jobs'
+    DjMetrics.instance.register_metrics_for_delayed_job_worker!
 
-        # Exit, ignoring signal handlers which would prevent the process from dying
-        exit!(0)
-      end
+    rescue_from StandardError do |err|
+      DjMetrics.instance.dj_job_status_total_metric.increment(labels: { queue: queue_name, priority: priority, status: 'failure', job_name: self.class.name })
+      raise err
     end
-  end
-
-  if ENV['ECS'] != 'true'
-    # When called through Delayed::Job, uses this hook
-    def before job
-      return unless STARTING_PATH != expected_path || ! File.exist?('config/exception_notifier.yml')
 
-      job = self unless job.respond_to? :locked_by
-
-      msg = "Started dir is `#{STARTING_PATH}`\nCurrent dir is `#{expected_path}`\nExiting in order to let systemd restart me in the correct directory."
-      notify_on_restart(msg)
-      unlock_job!(job.id)
-
-      # Exit, ignoring signal handlers which would prevent the process from dying
-      exit!(0)
+    # When called through Active::Job, uses this hook
+    before_perform do |job|
+      before_handler(job)
     end
-  end
 
-  private
-
-  def notify_on_restart msg
-    Rails.logger.info msg
-    return unless File.exist?('config/exception_notifier.yml')
-
-    setup_notifier('DelayedJobRestarter')
-    @notifier.ping(msg) if @send_notifications
-  end
-
-  def expected_path
-    Rails.cache.fetch('deploy-dir') do
-      # A default for the first deployment and local development
-      # This should be set on deployment.
-      Delayed::Worker::Deployment.deployed_to
```

Comment:
Calls to `deployed_to` or setting of the variable additionally appear in:
```
config/deploy.rb
config/initializers/delayed_job.rb
lib/tasks/cas.rake
```

If we're removing this concept, we should make sure it gets pulled out of those locations as well.

Reply:
Sounds good. I'll do that.